### PR TITLE
Altinn2 list endpoints, external revoke endpoints, attribute resolver, asserter framework 

### PR DESCRIPTION
**Pull request by andreasisnes**
## Description
* add Attribute resolver
* add model asserter framework
* add endpoint for altinn2 list offered and received delegations
* add external endpoints for revoking endpoints
* remove all warnings
* remove old dependencies

## Related Issue(s)
- #501 
- #598 
- #503 
- #577 

## Developer/Reviewer Checklist
- [x] **Your** code builds clean without any errors or warnings
- [x] No changes to config/appsettings or environment variables created in separate linked PR
- [x] Manual testing done (required)
- [x] Relevant integration test added (required for functional changes)
- [ ] Relevant automated test added (required for functional changes)
- [x] All integration tests run green

## Documentation
- [ ] User documentation is updated with a separate linked PR in [altinn-studio-docs.](https://github.com/Altinn/altinn-studio-docs) (if applicable)

### CODE REVIEW

**Review by jonkjetiloye — COMMENTED** [review]
Added some comments and thoughts. 
Let's have a chat tomorrow and "git er done"

File: Altinn3.ruleset [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <RuleSet Name="New Rule Set" Description=" " ToolsVersion="15.0">
   <Rules AnalyzerId="StyleCop.Analyzers" RuleNamespace="StyleCop.Analyzers">
-    <Rule Id="SA0001" Action="Error" />
+    <Rule Id="SA0001" Action="Warning" />
```
[the remark this answers is not in the document]
Comment (andreasisnes) [reply Altinn3.ruleset:4]: yes, like crazy :+1:

File: src/Altinn.AccessManagement.Persistence/DelegationMetadataRepository.cs [hunk 2]
```diff
@@ -245,6 +229,89 @@ public async Task<List<DelegationChange>> GetReceivedResourceRegistryDelegations
         }
     }
 
+    /// <inheritdoc/>
+    public async Task<List<DelegationChange>> GetReceivedDelegations(List<int> offeredByPartyIds, CancellationToken cancellationToken = default)
```
[the remark this answers is not in the document]
Comment (andreasisnes) [reply src/Altinn.AccessManagement.Persistence/DelegationMetadataRepository.cs:233]: hehehehe, done! Terminology simply dont fit.

image

File: src/Altinn.AccessManagement.Core/Services/Altinn2RightsService.cs [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+            {
+                offeredBy.AddRange(keyRoles);
+            }
+
+            return await _delegationRepository.GetReceivedDelegations(offeredBy, cancellationToken);
+        }
+
+        if (reportee.Id == AltinnXacmlConstants.MatchAttributeIdentifiers.OrganizationNumberAttribute)
+        {
+            var party = await _contextRetrievalService.GetPartyForOrganization(reportee.Value);
```
[the remark this answers is not in the document]
Comment (andreasisnes) [reply src/Altinn.AccessManagement.Core/Services/Altinn2RightsService.cs:110]: yeah, I know.. Seems a bit clumsy, bit hard to streamline the logic here, but as I understand the entity should be cached?

File: src/Altinn.AccessManagement.Core/Services/SingleRightsService.cs [hunk 4]
```diff
@@ -158,154 +160,52 @@ public async Task<DelegationActionResult> DelegateRights(int authenticatedUserId
...
-                return result;
+                return assertion;
             }
 
-            List<RequestToDelete> policiesToDelete = DelegationHelper.GetRequestToDeleteResourceRegistryService(authenticatedUserID, resource.Identifier, fromParty.PartyId, int.Parse(to.First().Value));
+            var toParty = await _resolver.Resolve(delegation.To, Urn.InternalIds, cancellationToken);
+            var fromParty = await _resolver.Resolve(delegation.From, Urn.InternalIds, cancellationToken);
+            var resource = await _resolver.Resolve(delegation.Rights?.FirstOrDefault()?.Resource ?? [], [Urn.Altinn.Resource.ResourceRegistryId], cancellationToken);
+
+            var policiesToDelete = DelegationHelper.GetRequestToDeleteResourceRegistryService(authenticatedUserId, resource.GetRequiredString(Urn.Altinn.Resource.ResourceRegistryId), fromParty.GetRequiredInt(Urn.InternalIds), toParty.GetRequiredInt(Urn.InternalIds));
```
[the remark this answers is not in the document]
Comment (jonkjetiloye) [reply src/Altinn.AccessManagement.Core/Services/SingleRightsService.cs:187]: Will register a bug for this which can be resolved separately

**Review by jonkjetiloye — APPROVED** [review]
Great work! 🥇 
As discussed I'll create follow up tasks for the remaining minor issues/bugs and follow up on those separately.

File: src/Altinn.AccessManagement.Core/Services/Altinn2RightsService.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (jonkjetiloye) [reply src/Altinn.AccessManagement.Core/Services/Altinn2RightsService.cs]: Can be resolved through #614

**Pull request merged**